### PR TITLE
Fix #46 - Bump dependency version

### LINKED ISSUE #46: Bump `omniauth-oauth2` dependency version
I'm using multiple oauth2 gems (google, ms..). These all have dependency omniauth-oauth2 (>= 1.3.1) and currently I'm on 1.4 so this leads to conflicts in the bundle when adding omniauth-slack. 

We'd like to use your gem but don't want to downgrade our version of omniauth-oauth2, because that would mean we would never be able to upgrade again.

@kmrshntr , would you mind bumping the dependency to include version 1.4 ('>= 1.3.1' / '~> 1.4.0') ?

### PR DESCRIPTION
**Pull request by jonhue**
Bumping up `omniauth-oauth2` to the most current version `1.4.0` to ensure compatibility with other omniauth solutions.
kmrshntr/omniauth-slack#46

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-source 'https://rubygems.org'
+source 'http://rubygems.org'
```
Comment (kmrshntr) [inline Gemfile]: Why did you change here? I think we should use https unless reasonable excuse.

Comment (jonhue) [inline Gemfile]: Loading gems from rubygems via SSL does not work for me. But sure, I can change that.

Comment (jonhue) [inline Gemfile]: I commited the changes.

Comment (kmrshntr) [inline Gemfile]: Thank you 💯

**Comment by jonhue** [comment]
@kmrshntr Do you release that fix as a new version?

**Comment by kmrshntr** [comment]
@jonhue Actually I hesitate release newer version recently because current master version doesn't work properly in some cases. I'm worried that causes overflowing of issues. But I have not figured out the problem in detail. Sorry for the inconvenience for now.

**Comment by kmrshntr** [comment]
I'll make time to solve this problem in a few weeks.

**Comment by jonhue** [comment]
@kmrshntr Ok, that's fine. I'll then just use a git repository as reference for now.

**Comment by ghost** [comment]
I am having the same issue the omniauth-quickbooks-oauth2 depends on omniauth-oauth2 version 1.5.1
and omniauth-slack depends on omniauth-oauth2 version 1.3.1. 
what should I do so that I can use both the gems.
@hare Krishna
